source/code/troubleshooter/omsagent_tsg_zip/tsg_files/tsg_code/tsg_general.py

```diff
@@ -0,0 +1,191 @@
+        "4: Agent consuming high CPU/memory.\n"\
+        "5: Installation failures.\n"\
+        "6: Custom logs issue.\n"\
+        "================================================================================\n"\
+        "A: Run through all scenarios.\n"\
+        "L: Collect the logs for OMS Agent.\n"\
+        "Q: Press 'Q' to quit.\n"\
+        "================================================================================")
+    switcher = {
+        '1': check_heartbeat,
+        '2': check_connection,
+        '3': check_syslog,
+        '4': check_high_cpu_memory,
+        '5': check_installation,
+        '6': check_custom_logs,
+        'A': check_all
+    }
+    issue = get_input("Please select an option",\
+                      (lambda x : x in ['1','2','3','4','5','6','q','quit','a','l']),\
+                      "Please enter an integer corresponding with your issue (1-6) to\n"\
+                        "continue (or 'A' to run through all scenarios), 'L' to run the log\n"\
+                        "collector, or 'Q' to quit.")
+    # quit troubleshooter
+    if (issue.lower() in ['q','quit']):
+        print("Exiting the troubleshooter...")
+        return
+    # collect logs
+    if (issue.lower() == 'l'):
+        print("Running the OMS Log Collector...")
+        print("================================================================================")
+        collect_logs()
+        return
+    # run troubleshooter
+    section = switcher.get(issue.upper(), lambda: "Invalid input")
+    print("================================================================================")
+    success = section()
+
+    print("================================================================================")
+    print("================================================================================")
+    # print out all errors/warnings
+    if (len(err_summary) > 0):
+        print("ALL ERRORS/WARNINGS ENCOUNTERED:")
+        for err in err_summary:
+            print("  {0}".format(err))
+        print("--------------------------------------------------------------------------------")
+        
+    # no errors found
+    if (success == 0):
+        print("No errors were found.")
+    # user requested to exit
+    elif (success == 1):
+        return
+    # error found
+    else:
+        print("Please review the errors found above.")
+    # give information to user about next steps
+    print("If you still have an issue, please run the troubleshooter again and collect the\n"\
+        "logs for OMS.\n"\
+        "In addition, please include the following information:\n"\
+        "  - Azure Subscription ID where the Log Analytics Workspace is located\n"\
+        "  - Workspace ID the agent has been onboarded to\n"\
+        "  - Workspace Name\n"\
+        "  - Region Workspace is located\n"\
+        "  - Pricing Tier assigned to the Workspace\n"\
+        "(The above points can all be found in Azure Support Center.)\n"\
```

Comment:
Not sure what this means to the customer. I thought ASC is an internal tool just for us and CSS? Or is this also the name for the portal where customers go to file a ticket?

Reply:
Apologies you're right, I originally took this from some documentation meant for CSS but forgot to change that, changed below:
https://github.com/microsoft/OMS-Agent-for-Linux/blob/bff7c80f869d6f5a4046ed41b4bd0ed89d69219f/source/code/troubleshooter/modules/main.py#L240-L249